app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryListFragment.java

```diff
@@ -0,0 +1,306 @@
+        // To combine lists and handle duplicates automatically
+        LinkedHashSet<String> userDictionaryLanguage = new LinkedHashSet<>();
+
+        // Add "For all languages"
+        userDictionaryLanguage.add("");
+
+        // Add the language from the user dictionary if a word is present
+        if (enabledUserDictionary != null) {
+            userDictionaryLanguage.addAll(enabledUserDictionary);
+        }
+
+        // Add the main language selected in the "Language and Layouts" setting
+        for (InputMethodSubtype mainSubtype : enabledMainSubtype) {
+            Locale mainLocale = LocaleUtils.constructLocaleFromString(mainSubtype.getLocale());
+            String mainLocaleString = mainLocale.toString();
+            // Special treatment for the known languages with _ZZ and _zz types
+            if (mainLocaleString.endsWith("_ZZ") || mainLocaleString.endsWith("_zz") || mainLocaleString.equals("zz")) {
+                mainLocaleString = LocaleUtils.getLocaleDisplayNameInSystemLocale(mainLocale, requireActivity()).toLowerCase();
+            }
+            userDictionaryLanguage.add(mainLocaleString);
+
+            // Add secondary language only if main language is selected and "Use system languages" setting is not enabled
+            for (Locale secondSubtype : Settings.getSecondaryLocales(prefs, String.valueOf(mainLocale))) {
+                Locale secondLocale = LocaleUtils.constructLocaleFromString(String.valueOf(secondSubtype));
+                String secondLocaleString = secondLocale.toString();
+                // Special treatment for the known languages with _ZZ and _zz types
+                if (secondLocaleString.endsWith("_ZZ") || secondLocaleString.endsWith("_zz") || secondLocaleString.equals("zz")) {
+                    secondLocaleString = LocaleUtils.getLocaleDisplayNameInSystemLocale(secondLocale, requireActivity()).toLowerCase();
+                }
+                if (!localeSystemOnly) {
+                    userDictionaryLanguage.add(secondLocaleString);
+                }
+            }
+        }
+
+        // Finally, add the enabled system languages
+        for (Locale subtype : enabledSystemLocale) {
+            Locale systemLocale = LocaleUtils.constructLocaleFromString(String.valueOf(subtype));
+            String systemLocaleString = systemLocale.toString();
+            // Special treatment for the known languages with _ZZ and _zz types
+            if (systemLocaleString.endsWith("_ZZ") || systemLocaleString.endsWith("_zz") || systemLocaleString.equals("zz")) {
+                systemLocaleString = LocaleUtils.getLocaleDisplayNameInSystemLocale(systemLocale, requireActivity()).toLowerCase();
+            }
+            userDictionaryLanguage.add(systemLocaleString);
+        }
+
+        // Sort languages alphabetically
+        ArrayList<String> sortedLanguages = new ArrayList<>(userDictionaryLanguage);
+        Collections.sort(sortedLanguages, String::compareToIgnoreCase);
```

Comment:
When you want to have sorted language strings, you can just start with a `TreeSet` (with comparator) instead of a `LinkedHashSet`. Then it's sorted already.
`LinkedHashSet` is useful if you want to preserve insertion order, but you don't seem to do that.